src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ComCreateCaseloadService.kt

```diff
@@ -55,36 +54,39 @@ class ComCreateCaseloadService(
 
   private fun buildCreateCaseload(managedOffenders: List<ManagedOffenderCrn>): List<ComCase> {
     val deliusAndNomisRecords = pairDeliusRecordsWithNomis(managedOffenders)
-    val crnsToLicenceLists = mapCasesToLicences(deliusAndNomisRecords)
-    val eligibleCases = filterCasesEligibleForCvl(deliusAndNomisRecords, crnsToLicenceLists)
+    val nomisIdsToAreaCodes = deliusAndNomisRecords.map { (deliusRecord, nomisRecord) -> nomisRecord.prisonerNumber to (deliusRecord.team?.provider?.code ?: "") }.toMap()
+    val cvlRecords = cvlRecordService.getCvlRecords(deliusAndNomisRecords.map { (_, nomisRecord) -> nomisRecord }, nomisIdsToAreaCodes)
+    val crnsToLicenceLists = mapCasesToLicences(deliusAndNomisRecords, cvlRecords)
+    val eligibleCases = filterCasesEligibleForCvl(deliusAndNomisRecords, crnsToLicenceLists, cvlRecords)
```

Comment:
This is now pretty dense and difficult to read, I think because it's mixing up different levels of abstraction. 

Maybe we just leave it for now and look at tidying up afterwards